store/sqlstore/role_supplier.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type Role struct {
+	Id            string
+	Name          string
+	DisplayName   string
+	Description   string
+	Permissions   string
+	SchemeManaged bool
+}
+
+func FromRole(role *model.Role) *Role {
+	return &Role{
+		Id:            role.Id,
+		Name:          role.Name,
+		DisplayName:   role.DisplayName,
+		Description:   role.Description,
+		Permissions:   strings.Join(role.Permissions, " "),
```

Comment:
I expect (maybe not here, but anywhere) to have some kind of "Set" behavior. I mean, IMO, we must avoid repeated permissions in the same role.